ecommerce/views/v0/__init__.py

```diff
@@ -425,110 +423,80 @@ def post(self, request, *args, **kwargs):
         clear out the stored basket)
         3. Perform any enrollments, account status changes, etc.
         """
-        if not PaymentGateway.validate_processor_response(
-            ECOMMERCE_DEFAULT_PAYMENT_GATEWAY, request
-        ):
-            raise PermissionDenied(
-                "Could not validate response from the payment processor."
-            )
 
-        payment_data = request.POST
-
-        processor_response = PaymentGateway.get_formatted_response(
-            ECOMMERCE_DEFAULT_PAYMENT_GATEWAY, request
-        )
-        converted_order = PaymentGateway.get_gateway_class(
-            ECOMMERCE_DEFAULT_PAYMENT_GATEWAY
-        ).convert_to_order(payment_data)
-
-        order_id = Order.decode_reference_number(converted_order.reference)
-
-        try:
-            order = Order.objects.get(pk=order_id)
-        except ObjectDoesNotExist:
+        order = api.get_order_from_cybersource_payment_response(request)
+        if order == None:
```

Comment:
When comparing to `None` you need to use `is` as it's a singleton value.
```suggestion
        if order is None:
```